TESTs/explore.integration.test.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import Explore from '@/app/explore/page';
+
+// Mock the fetchData function to simulate API call
+import { fetchData } from '../utils/fetchData';
+
+describe('Explore Component', () => {
```

Comment:
Using mocks there decouples the explorer component behavior from the backend system making this more of a unit test then an integration test since you are not testing how the two components work together but rather testing a single component in isolation.